File: app/src/main/java/com/indisparte/pothole/di/component/PotholeApplication.java
package com.indisparte.pothole.di.component;

import android.app.Application;
import android.content.Context;

import dagger.hilt.android.HiltAndroidApp;

/**
 * @author Antonio Di Nuzzo (Indisparte)
 */
@HiltAndroidApp
public class PotholeApplication extends Application {
    private static Context sApplicationContext;

    @Override
    public void onCreate() {
        super.onCreate();

        sApplicationContext = getApplicationContext();

    }

    public static Context getContext() {
        return sApplicationContext;
    }

}

File: app/src/main/java/com/indisparte/pothole/di/component/PotholeApplication.kt
package com.indisparte.pothole.di.component

import android.app.Application
import android.content.Context
import dagger.Provides
import dagger.hilt.android.HiltAndroidApp

/**
 * @author Antonio Di Nuzzo (Indisparte)
 */
@HiltAndroidApp
class PotholeApplication : Application() {
    // No need to override onCreate()

    // Declare a @Provides method for the Context dependency
    @Provides
    fun provideContext(): Context = applicationContext
}

